1-ECommerce/src/components/Products.tsx

```diff
@@ -0,0 +1,91 @@
+import { useNavigate, useSearchParams } from "react-router-dom";
+
+import { Button } from "./Button";
+import { NothingToShow } from "./NothingToShow";
+import { ProductCard } from "./ProductCard";
+import type { Product } from "./ProductsContainer";
+
+interface ProductProps {
+  products: Product[];
+  favorites: Product[];
+  onFavoriteClick: (product: Product) => void;
+}
+
+type SortProductFunction = (a: Product, b: Product) => number;
+
+const sortAZ = (a: Product, b: Product): number =>
+  a.title.localeCompare(b.title, "en", { sensitivity: "base" });
+
+const sortZA = (a: Product, b: Product): number =>
+  b.title.localeCompare(a.title, "en", { sensitivity: "base" });
+
+const sortLowPrice = (a: Product, b: Product): number => b.price - a.price;
+const sortHighPrice = (a: Product, b: Product): number => a.price - b.price;
+
+const sortingFunctions = {
+  az: sortAZ,
+  za: sortZA,
+  lowPrices: sortLowPrice,
+  highPrices: sortHighPrice,
+};
+
+export const Products = ({
+  products,
+  favorites,
+  onFavoriteClick,
+}: ProductProps) => {
+  const [queryParams, setQueryParams] = useSearchParams();
+  const navigate = useNavigate();
+
+  const API_TOTAL_PRODUCTS = 100;
+  const PRODUCTS_PER_PAGE = 6;
+  const currentLimit = Number(queryParams.get("limit") ?? 6);
+
+  const view = String(queryParams.get("view"));
+
+  const sortedProducts =
+    view in sortingFunctions
+      ? products.toSorted(sortingFunctions[view] as SortProductFunction)
+      : products;
+
+  const isLoadMoreVisible =
+    products.length > 0 && currentLimit < API_TOTAL_PRODUCTS;
+
+  const handleLoadMoreClick = () => {
+    const newLimit =
+      currentLimit + PRODUCTS_PER_PAGE < API_TOTAL_PRODUCTS
+        ? currentLimit + PRODUCTS_PER_PAGE
+        : API_TOTAL_PRODUCTS;
+
+    queryParams.set("limit", JSON.stringify(newLimit));
+    setQueryParams(queryParams);
```

Comment:
The code could be an option, but I don't like having the `setQueryParams` be an impure function (but I don't know if there's another way)

`setQueryParams((params) => {
   params..set("limit", JSON.stringify(newLimit));
   return params;
 })`

Reply:
I added `limit` the the query params to trigger the effect (since it has `location = useLocation()` as a dependency it will run whenever the url changes). Another option would be to define `limit` inside the `useFilteredProducts` hook and also return `limit` and `setLimit` from that hook (and add `limit` as a dependency in the effect). Something I like about this is that I wont have to worry about parsing the parameter from the url, and I can work with it being sure that it's a number. Does this idea sound ok?

Reply:
Let's discuss it in the daily